core/qtserialbluetooth.cpp

```diff
@@ -295,45 +196,51 @@ static dc_status_t qt_serial_write(void *io, const void* data, size_t size, size
 
 		nbytes += rc;
 	}
-#endif
+
 	if (actual)
 		*actual = nbytes;
 
 	return DC_STATUS_SUCCESS;
 }
 
-static dc_status_t qt_serial_purge(void *io, dc_direction_t)
+static dc_status_t qt_serial_poll(void *io, int timeout)
 {
 	qt_serial_t *device = (qt_serial_t*) io;
 
-	if (device == NULL)
+	if (!device)
 		return DC_STATUS_INVALIDARGS;
-#if !defined(Q_OS_WIN)
-	if (device->socket == NULL)
+
+	if (!device->socket)
 		return DC_STATUS_INVALIDARGS;
-#endif
-	// TODO: add implementation
+	if (device->socket->waitForReadyRead(timeout))
```

Comment:
i was checking if Qt for Windows has implementation for this and found:

https://doc.qt.io/qt-5/qbluetoothsocket.html

> Note: QBluetoothSocket does not support synchronous read and write operations. Functions such as waitForReadyRead() and waitForBytesWritten() are not implemented. I/O operations should be performed using readyRead(), read() and write().

there is also this comment in the underlying (abstract) implementation for Windows:

https://doc.qt.io/qt-5/qabstractsocket.html#waitForReadyRead

> Note: This function may fail randomly on Windows. Consider using the event loop and the readyRead() signal if your software will run on Windows.

so i think we should be doing something similar to what we have already here:
https://github.com/Subsurface-divelog/subsurface/blob/e28be280cd73fce5f41215ccb476ce97bc4073a7/core/qtserialbluetooth.cpp#L152

```
static dc_status_t qt_serial_poll(void *io, int timeout)
{
	qt_serial_t *device = (qt_serial_t*) io;

	if (!device)
		return DC_STATUS_INVALIDARGS;

	if (!device->socket)
		return DC_STATUS_INVALIDARGS;

	QEventLoop loop;
	QTimer timer;
	timer.setSingleShot(true);
	loop.connect(&timer, SIGNAL(timeout()), SLOT(quit()));
	loop.connect(device->socket, SIGNAL(readyRead()), SLOT(quit()));
	timer.start(device->timeout);
	loop.exec();
	
	if (!timer.isActive())
		return DC_STATUS_SUCCESS;
	return DC_STATUS_TIMEOUT;
}
```

Reply:
i can send a follow up for this once we resolve the build problem for AppImage.

Reply:
Let's ignore this for the moment and assume that it will work once merged...
You should be able to push additional commits into the other PR

Reply:
sorry, got confused. THIS PR

Reply:
i pushed a commit.